### PR TITLE
Bump 4.5 changes and align master as 4.6 branch.

### PR DESCRIPTION
**Pull request by fedepaol**
This includes removing the sctptester image as it's being consumed directly from quay mirror.

### CODE REVIEW

File: ci-operator/config/openshift-kni/cnf-features-deploy/openshift-kni-cnf-features-deploy-master.yaml [hunk 1]
```diff
@@ -53,26 +53,23 @@ resources:
       memory: 200Mi
 tag_specification:
   cluster: https://api.ci.openshift.org
-  name: "4.5"
+  name: "4.6"
```
Comment (slintes) [inline ci-operator/config/openshift-kni/cnf-features-deploy/openshift-kni-cnf-features-deploy-master.yaml:56]: fyi: at least 2 days ago the 4.6 didn't work yet, we might need to keep 4.5 for now (did the same on the operator repo)

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay [hunk 2]
```diff
@@ -9,3 +9,4 @@ registry.svc.ci.openshift.org/openshift-kni/4.5:sctptester quay.io/openshift-kni
 registry.svc.ci.openshift.org/openshift-kni/4.5:cnftest-utils quay.io/openshift-kni/cnftest-utils:4.5 quay.io/openshift-kni/cnftest-utils:4.5.0 quay.io/openshift-kni/cnftest-utils:latest
 registry.svc.ci.openshift.org/openshift-kni/4.5:stresser quay.io/openshift-kni/stresser:4.5 quay.io/openshift-kni/stresser:4.5.0 quay.io/openshift-kni/stresser:latest
 registry.svc.ci.openshift.org/openshift-kni/4.5:dpdk quay.io/openshift-kni/dpdk:4.5 quay.io/openshift-kni/dpdk:4.5.0 quay.io/openshift-kni/dpdk:latest
+registry.svc.ci.openshift.org/openshift-kni/4.6:cnf-tests quay.io/openshift-kni/cnf-tests:4.6 quay.io/openshift-kni/cnf-tests:4.6.0 quay.io/openshift-kni/cnf-tests:latest
```
Comment (slintes) [inline core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:12]: - `cnf-tests:latest` will be mirrored from 4.5.0 and 4.6.0 now, the last one wins. I don't think this is wanted?
- and what about the other images?

Maybe it's best to switch from using `latest` tag to versioned tags?

Comment (fedepaol) [reply core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:12]: >     * `cnf-tests:latest` will be mirrored from 4.5.0 and 4.6.0 now, the last one wins. I don't think this is wanted?

Not wanted at all!
> 
>     * and what about the other images?
> 
The others are used for tests. For now I'd use the 4.5 version even in 4.6 tests, at least for the initial phase.

> 
> Maybe it's best to switch from using `latest` tag to versioned tags?

Where we use the images we are already pinning to 4.5, if this is what you mean.

Comment (slintes) [reply core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:12]: > For now I'd use the 4.5 version even in 4.6 tests
> we are already pinning to 4.5

ah, I missed that we are pushing both latest and versioned images already. Just using 4.5. for now works for me :+1:

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay [hunk 3]
```diff
@@ -9,3 +9,4 @@ registry.svc.ci.openshift.org/openshift-kni/4.5:sctptester quay.io/openshift-kni
 registry.svc.ci.openshift.org/openshift-kni/4.5:cnftest-utils quay.io/openshift-kni/cnftest-utils:4.5 quay.io/openshift-kni/cnftest-utils:4.5.0 quay.io/openshift-kni/cnftest-utils:latest
```
Comment (SchSeba) [inline core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:9]: Can we also bump this one too?

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay [hunk 4]
```diff
@@ -9,3 +9,4 @@ registry.svc.ci.openshift.org/openshift-kni/4.5:sctptester quay.io/openshift-kni
...
 registry.svc.ci.openshift.org/openshift-kni/4.5:stresser quay.io/openshift-kni/stresser:4.5 quay.io/openshift-kni/stresser:4.5.0 quay.io/openshift-kni/stresser:latest
```
Comment (SchSeba) [inline core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:10]: Can we also bump this one too?

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay [hunk 5]
```diff
@@ -9,3 +9,4 @@ registry.svc.ci.openshift.org/openshift-kni/4.5:sctptester quay.io/openshift-kni
...
 registry.svc.ci.openshift.org/openshift-kni/4.5:dpdk quay.io/openshift-kni/dpdk:4.5 quay.io/openshift-kni/dpdk:4.5.0 quay.io/openshift-kni/dpdk:latest
```
Comment (SchSeba) [inline core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:11]: Can we also bump this one too?

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay (hunk 3 above)
Comment (fedepaol) [reply core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:9]: For test images I'd wait until there's real need to use a different version in 4.6

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay (hunk 4 above)
Comment (fedepaol) [reply core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:10]: For test images I'd wait until there's real need to use a different version in 4.6

File: core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay (hunk 5 above)
Comment (fedepaol) [reply core-services/image-mirroring/openshift-kni/mapping_openshift-kni_quay:11]: For test images I'd wait until there's real need to use a different version in 4.6

**Comment by fedepaol** [comment]
@slintes @SchSeba so, master still points to 4.5
I left the mirroring of 4.6 to better times, when the 4.6 stream will be available.

On master, I disable the promotion of 4.5 images so we are sure they come from the 4.5 branch.
I also spotted that the sctptester was being built twice from two different dockerfiles, I am going to remove the leftover from the cnf-features-deploy repo

File: ci-operator/config/openshift-kni/cnf-features-deploy/openshift-kni-cnf-features-deploy-master.yaml [hunk 6]
```diff
@@ -44,6 +40,12 @@ images:
   from: base
   to: dpdk
 promotion:
+  excluded_images:
```
Comment (slintes) [inline ci-operator/config/openshift-kni/cnf-features-deploy/openshift-kni-cnf-features-deploy-master.yaml:43]: when I see correct, all images are excluded now? I know this has been an issue, not sure if it was fixed. If something does not work as expected, you might need to remove the promotion section.

Comment (fedepaol) [reply ci-operator/config/openshift-kni/cnf-features-deploy/openshift-kni-cnf-features-deploy-master.yaml:43]: If I remove the promotion sections all the images are promoted anyway using the tag_specification stream (see https://github.com/openshift/ci-tools/blob/master/CONFIGURATION.md#promotion) .

The issue here is that both master and release-4.5 are pointing to the 4.5 stream, so images would be promoted from two different branches (and also, CI was smart enough to raise that as an error, so I had to add this).

Comment (slintes) [reply ci-operator/config/openshift-kni/cnf-features-deploy/openshift-kni-cnf-features-deploy-master.yaml:43]: oh, good to know there is some default promotion, missed it :+1:

**Pull request merged**